lib/logrecord.js

```diff
@@ -23,8 +27,15 @@ LogRecord.prototype.getFormattedMeta = function () {
   return out.join(',')
 }
 
-
 LogRecord.prototype.getFormattedMessage = function () {
+  if (!this._formattedMessage) {
+    this._formattedMessage = this._getFormattedMessage()
+  }
+  return this._formattedMessage
+}
+
+
+LogRecord.prototype._getFormattedMessage = function () {
```

Comment:
Agreed -- but in this case where the current iteration of `getFormattedMessage` has multiple return paths, this seemed slightly more readable.  I'm happy to change if you would like.